stdlib/uuid.pyi

```diff
@@ -72,10 +73,10 @@ class UUID:
     def version(self) -> _Int | None: ...
     def __int__(self) -> _Int: ...
     def __eq__(self, other: Any) -> bool: ...
-    def __lt__(self, other: Any) -> bool: ...
-    def __le__(self, other: Any) -> bool: ...
-    def __gt__(self, other: Any) -> bool: ...
-    def __ge__(self, other: Any) -> bool: ...
+    def __lt__(self, other: SupportsDunderLT) -> bool: ...
+    def __le__(self, other: SupportsDunderLE) -> bool: ...
+    def __gt__(self, other: SupportsDunderGT) -> bool: ...
+    def __ge__(self, other: SupportsDunderGE) -> bool: ...
```

Comment:
Oh, you are right. Types are not really correct. But, I guess not in a way you've described.

Consider this example:

```python
>>> class CanCompare:
...    def __gt__(self, other):
...       return isinstance(other, uuid.UUID)
... 
>>> uuid.uuid4() < CanCompare()
True
```

UUID can be compared with non-UUID types. But, I have to change `LT` -> `GT`

Reply:
The `<` operator works between instances of `UUID` and `CanCompare`, but in the stubs we represent that by having the `other` argument of `CanCompare.__gt__` be `Any`, rather than complicating `UUID.__lt__`.

This is why type-checkers recognise `<` as being supported between `int` and `float`. `int.__lt__` is annotated as only accepting `int`s, but `float.__gt__` is annotated as accepting `float`s. Since `int` is a subtype of `float` for type-checkers, type-checkers know that `4 < 5.7` is supported because of the permissive type in `float.__gt__`, not because of the way `int.__lt__` is annotated.

Reply:
Ok, let's use `UUID` to be safe! 👍